gravitee-apim-console-webui/src/management/api/api-ng-navigation/api-ng-v4-menu.service.ts

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright (C) 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { Injectable } from '@angular/core';
+
+import { MenuGroupItem, MenuItem } from './MenuGroupItem';
+
+import { GioPermissionService } from '../../../shared/components/gio-permission/gio-permission.service';
+
+@Injectable()
+export class ApiNgV4MenuService {
```

Comment:
What about creating an interface/parent class to be used by the two menus? Could contain the permission service and the basic menu methods.

Reply:
Yes, added.